src/App.js

```diff
@@ -1,19 +1,37 @@
-import React from 'react';
+import React, {useState} from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const [messages, setMessages] = useState(chatMessages);
+
+  const handleLikeClick = (id) => {
+    setMessages((prevMessages) => {
+      const updatedMessages = prevMessages.map((entry) => {
+        if (entry.id === id) {
+          return {...entry, liked: !entry.liked };
+        } else {
+          return entry;
+        }
+      });
+      return updatedMessages;
+    });
+  };
+
+  const totalLikes = messages.filter((entry) => entry.liked).length;
```

Comment:
Filter usually implies that we are trying to filter something out by a condition. Prefer to use `reduce`, which will achieve the same thing, but is more descriptive of what we're trying to do which is to take many values and reduce it down to one summed value. Like:
```javascript
  const totalLikes = () => {
    return messages.reduce((total, entry) => {
      return entry.liked ? total + 1 : total;
    }, 0)
  };